src/pages/workspaces/workspace/notebooks/AnalysisLauncher.js

```diff
@@ -235,31 +245,49 @@ const PreviewHeader = ({
       [readOnlyAccess, () => h(HeaderButton, { onClick: () => setExportingAnalysis(true) },
         [makeMenuIcon('export'), 'Copy to another workspace']
       )],
-      //TODO: THIS CONDITIONAL should look for Jupyter or Rstudio
-      [!!runtimeStatus && runtime.labels.tool !== 'Jupyter', () => h(StatusMessage, { hideSpinner: true }, [
-        'Your cloud compute doesn\'t appear to be running Jupyter. Create a new cloud environment with Jupyter on it to edit this notebook.'
-      ])],
       [!mode || [null, 'Stopped'].includes(runtimeStatus), () => h(Fragment, [
```

Comment:
GitHub didn't let me make a suggestion selecting a large number of lines so I'll copy my styling suggestion below, with motivations being
- consistency with the rest of the codebase
- conserving vertical space when the above isn't violated

```javascript
  return h(ApplicationHeader, {
    label: 'PREVIEW (READ-ONLY)',
    labelBgColor: colors.dark(0.2)
  }, [
    Utils.cond(
      [readOnlyAccess, () => h(HeaderButton, { onClick: () => setExportingAnalysis(true) }, [
        makeMenuIcon('export'), 'Copy to another workspace'
      ])],
      [!mode || [null, 'Stopped'].includes(runtimeStatus), () => h(Fragment, [
        ...(toolLabel === tools.Jupyter.label ? [
          Utils.cond(
            [runtime && !welderEnabled, () => h(HeaderButton, { onClick: () => setEditModeDisabledOpen(true) }, [
              makeMenuIcon('warning-standard'), 'Edit (Disabled)'
            ])],
            [locked, () => h(HeaderButton, { onClick: () => setFileInUseOpen(true) }, [
              makeMenuIcon('lock'), 'Edit (In use)'
            ])],
            () => h(HeaderButton, {
              onClick: () => { currentRuntimeTool !== tools.Jupyter.label ? setCreateOpen(true) : chooseMode('edit') }
            }, [
              makeMenuIcon('edit'), 'Edit'
            ])
          ),
          h(HeaderButton, {
            onClick: () => getLocalPref('hidePlaygroundMessage') ? chooseMode('playground') : setPlaygroundModalOpen(true)
          }, [
            makeMenuIcon('chalkboard'), 'Playground mode'
          ])
        ] : [
          h(HeaderButton, {
            onClick: () => {
              currentRuntimeTool !== tools.RStudio.label ?
                setCreateOpen(true) :
                Nav.goToPath('workspace-application-launch', { namespace, name, application: 'RStudio' })
            },
            disabled: runtimeStatus !== 'Running',
            tooltip: runtimeStatus !== 'Running' ? 'You must have a running cloud environment' : ''
          }, [
            makeMenuIcon('rocket'), 'Launch'
          ])
        ]),
        h(MenuTrigger, {
          closeOnClick: true,
          content: h(Fragment, [
            h(MenuButton, { 'aria-label': `Copy analysis`, onClick: () => setCopyingAnalysis(true) }, ['Make a Copy']),
            h(MenuButton, { 'aria-label': `Export analysis`, onClick: () => setExportingAnalysis(true) }, ['Copy to another workspace']),
            h(MenuButton, {
              'aria-label': `Copy to clipboard`,
              onClick: withErrorReporting('Error copying to clipboard', async () => {
                await clipboard.writeText(`${window.location.host}/${analysisLink}`)
                notify('success', 'Successfully copied URL to clipboard', { timeout: 3000 })
              })
            }, ['Copy URL to clipboard'])
          ]),
          side: 'bottom'
        }, [
          h(HeaderButton, {}, [icon('ellipsis-v')])
        ])
      ])],
      [_.includes(runtimeStatus, usableStatuses), () => {
        console.assert(false, `Expected cloud environment to NOT be one of: [${usableStatuses}]`)
        return null
      }],
      [runtimeStatus === 'Creating', () => h(StatusMessage, [
        'Creating cloud environment. You can navigate away and return in 3-5 minutes.'
      ])],
      [runtimeStatus === 'Starting', () => h(StatusMessage, [
        'Starting cloud environment, this may take up to 2 minutes.'
      ])],
      [runtimeStatus === 'Stopping', () => h(StatusMessage, [
        'Cloud environment is stopping, which takes ~4 minutes. You can restart it after it finishes.'
      ])],
      [runtimeStatus === 'LeoReconfiguring', () => h(StatusMessage, [
        'Cloud environment is updating, please wait.'
      ])],
      [runtimeStatus === 'Error', () => h(StatusMessage, { hideSpinner: true }, ['Cloud environment error.'])]
    ),
    div({ style: { flexGrow: 1 } }),
    div({ style: { position: 'relative' } }, [
      h(Clickable, {
        'aria-label': 'Exit preview mode',
        style: { opacity: 0.65, marginRight: '1.5rem' },
        hover: { opacity: 1 }, focus: 'hover',
        onClick: () => Nav.goToPath('workspace-analyses', { namespace, name })
      }, [icon('times-circle', { size: 30 })])
    ]),
    editModeDisabledOpen && h(EditModeDisabledModal, {
      onDismiss: () => setEditModeDisabledOpen(false),
      onRecreateRuntime: () => {
        setEditModeDisabledOpen(false)
        onCreateRuntime()
      },
      onPlayground: () => {
        setEditModeDisabledOpen(false)
        chooseMode('playground')
      }
    }),
    fileInUseOpen && h(FileInUseModal, {
      namespace, name, lockedBy, canShare, bucketName,
      onDismiss: () => setFileInUseOpen(false),
      onCopy: () => {
        setFileInUseOpen(false)
        setCopyingAnalysis(true)
      },
      onPlayground: () => {
        setFileInUseOpen(false)
        chooseMode('playground')
      }
    }),
    copyingAnalysis && h(AnalysisDuplicator, {
      printName: getDisplayName(analysisName),
      toolLabel: getTool(analysisName),
      fromLauncher: true,
      wsName: name, googleProject, namespace, bucketName, destroyOld: false,
      onDismiss: () => setCopyingAnalysis(false),
      onSuccess: () => setCopyingAnalysis(false)
    }),
    exportingAnalysis && h(ExportAnalysisModal, {
      printName: getDisplayName(analysisName),
      toolLabel: getTool(analysisName), workspace,
      fromLauncher: true,
      onDismiss: () => setExportingAnalysis(false)
    }),
    playgroundModalOpen && h(PlaygroundModal, {
      onDismiss: () => setPlaygroundModalOpen(false),
      onPlayground: () => {
        setPlaygroundModalOpen(false)
        chooseMode('playground')
      }
    })
  ])
```

Reply:
I implemented most of it, save for one suggested ternary. I like using if/else over a ternary when there are heavy side effects (in this case one branch navigates away from the page entirely)

Reply:
That makes sense. 

Sorry for the painful manual diff on my comment. I should've made a PR into your branch to make it easier to realize what I had suggested changing.